js/tests/test_examples.test.js

```diff
@@ -0,0 +1,109 @@
+/**
+ * E2E Tests for plugin demos and examples.
+ */
+
+const { execFile } = require('child_process');
+const path = require('path');
+const { promisify } = require('util');
+const execFileAsync = promisify(execFile);
+
+// Paths
+const PLUGINS = path.resolve(__dirname, '../plugins');
+const EXAMPLES_PATH = path.resolve(__dirname, '../examples');
+
+// Require env vars
+const { OPENAI_API_KEY, COMPOSIO_API_KEY, JULEP_API_KEY, JULEP_API_URL, LISTENNOTES_API_KEY } = process.env;
+console.log(COMPOSIO_API_KEY);
+
+// Plugin test definitions
+const EXAMPLES = {
```

Comment:
Consider adding JSDoc type definitions for the EXAMPLES object structure to improve code documentation and maintainability. Example:
```javascript
/**
 * @typedef {Object} TestExample
 * @property {string} plugin - The name of the plugin being tested
 * @property {string} file - Path to the test file
 * @property {Object} match - Output matching criteria
 * @property {('stdout'|'stderr')} match.type - Type of output to match
 * @property {string[]} match.values - Values to match in the output
 * @property {Object} env - Environment variables required for the test
 * @property {string} [cwd] - Optional working directory for the test
 */
```